extensions/git/src/commands.ts

```diff
@@ -1787,8 +1787,17 @@ export class CommandCenter {
 			ignoreFocusOut: true,
 			validateInput: (name: string) => {
 				const validateName = new RegExp(branchValidationRegex);
-				if (validateName.test(sanitize(name))) {
-					return null;
+				const sanitizedName = sanitize(name);
+				if (validateName.test(sanitizedName)) {
+					// If the sanitized name that we will use is different than what is
+					// in the input box, show an info message to the user informing them
+					// the branch name that will be used.
+					return name === sanitizedName
+						? null
+						: {
+							message: localize('branch name does not match sanitized', "Will be created as '{0}'", sanitizedName),
```

Comment:
Could we update the wording so that it says "The new branch will be..."? Thanks!

Reply:
```suggestion
							message: localize('branch name does not match sanitized', "The new branch will be '{0}'", sanitizedName),
```